runbot_gitlab/models/runbot_build.py

```diff
@@ -0,0 +1,90 @@
+# -*- encoding: utf-8 -*-
+##############################################################################
+#
+#    Odoo, Open Source Management Solution
+#    This module copyright (C) 2010 Savoir-faire Linux
+#    (<http://www.savoirfairelinux.com>).
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+from openerp import api, models, fields
+
+from .runbot_repo import escape_branch_name
+
+
+class runbot_build(models.Model):
+    _inherit = "runbot.build"
+
+    @api.depends('repo_id.uses_gitlab', 'branch_id', 'name')
+    def _compute_dest(self):
+        for build in self.filtered('repo_id.uses_gitlab'):
+            nickname = escape_branch_name(build.branch_id.name)
```

Comment:
I would suggest to truncate the `nickname` after 32 chars as in the original code because too long names can cause trouble with the database and the filesystem.